cddl/update-record.cddl

```diff
@@ -0,0 +1,12 @@
+update-triple-record = [
+&(subj: 0) => $subject-type-choice
+& (mval: 1) => [+measurement-values-map] ; object of the update
+? &(authorized-by: 2) => [ + $crypto-key-type-choice ]
+&(action: 3) => $update-action-type-choice
+]
+
+$subject-type-choice /= [+environment-map] ; One or more environment that comprise the subject
+$subject-type-choice /= [+stateful-environment-record] ;  One or more stateful-environments that comprise the subject
+
+$update-action-type-choice /= add ; The new triple causes a new revision identified by measurement-values-map , to be simply updated/added in Verifier
```

Comment:
@nedmsmith : The goal of `$update-action-type-choice /= add` is NOT to remove anything from the Verifier, but to increment.

However, as we are using a `common` `update-triple` for any Reference Value and Endorsed Value we  have two options:

(a) Specify what is that is been updated by specifying the type - RefVal/EndVal/CondEnd pertaining to the Subject

(b) Rely on the Supply Chain to exactly issue the same `triple` (like RefVal Triple) OR an EndVal Triple with the new `&(action: 3) => $update-action-type-choice` and 
indicate whether it is an `ADD` or a `REPLACE`

**Choice:**
Option-a: Verifier matches `subj` and based on the `type` attaches the new `triple` to the same `subj` 

Option - b: is `Neater` but implies modifying every triple where we need an update, to set the action

I would request we focus on the `intent` of the action and then agree whether to go with option `a` or option `b`

Reply:
>I would request we focus on the intent of the action and then agree whether to go with option a or option b

Maybe this part of the discussion should happen in issue #171 since the conclusion would affect the PR definition.